src/main/java/edu/slu/auxiliary/CompressorService.java

```diff
@@ -0,0 +1,142 @@
+ * Anything else has to be provided.  Keys to keep can be provided.  
+ */
+package edu.slu.auxiliary;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import net.sf.json.JSONObject;
+
+public class CompressorService{
+    
+    //These are keys that we want to make sure stay with the JSONObject.
+    //Think about things like "target" and "on"...are they special?  Current thinking is no, they are just as viable to be removed as any other property. 
+    
+    private Set<String> primitiveKeys = new HashSet<>(Arrays.asList("@id",
+        "id", 
+        "type", 
+        "@type", 
+        "label", 
+        "name", 
+        "summary", 
+        "description", 
+        "@context"
+    )); 
+        
+    
+    //These are keys that will be removed 100% of the time.  Maybe just __rerum.
+    private final String[] alwaysRemove = {
+        "__rerum"
+    };
+    
+    /**
+     * Default Initializer.  Probably just to use CompressorService.compress();
+     */
+    public CompressorService(){
+        
+    }
+    
+    /**
+     * Remove keys from a JSONObject that should not be kept because of compression.
+     * This will remove every key from the provided JSONObject that is not in the array of default primitive keys.
+     * @param toCompress A JSONObject to compress (by removing unwanted keys)
+     * @param keysToKeep A String[] Array of keys that should not be removed.
+     * @return toCompress once the keys are removed, or the original if there was an error.
+     */
+    public JSONObject compress(JSONObject toCompress){
+        Iterator keys = toCompress.keys();
+        JSONObject orig = JSONObject.fromObject(toCompress);
+        System.out.println("Compress A");
+        try{
+            while(keys.hasNext()){
+                String key = (String)keys.next();
+                if(!primitiveKeys.contains(key)){
+                    keys.remove();
+                }
+            }
+            return toCompress;
+        }
+        catch(Exception e){
+            System.out.println("Could not compress object A.  See error below.");
+            System.out.println(e.getStackTrace());
+            return orig;
+        }
+    }
+    
+    /**
+     * Remove keys from a JSONObject that should not be kept because of compression.
+     * This will remove every key from the provided JSONObject that is not in the provided array of keys to keep.
+     * @param toCompress A JSONObject to compress (by removing unwanted keys)
+     * @param keysToKeep A String[] Array of keys that should not be removed.
+     * @return toCompress once the keys are removed, or the original if there was an error.
+     */
+    public JSONObject compress(JSONObject toCompress, Set<String> keysToKeep){
+        keysToKeep.addAll(primitiveKeys);
+        Iterator<String> keys = toCompress.keySet().iterator();
```

Comment:
why not `keys()` like above? That would solve the casting to Object and the operation error you had below, I am fairly sure.

Reply:
:60